src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp

```diff
@@ -1186,37 +1186,70 @@ int xcldev::device::runTestCase(const std::string& py,
     std::string devInfoPath = name + "/test/";
     std::string xsaXclbinPath = getXsaPath(vendor) + devInfoPath;
     std::string dsaXclbinPath = dsaPath + devInfoPath;
-    std::string xrtTestCasePath = xrtPath + "test/" + py;
+    std::string xrtTestCasePath = xrtPath + "test/";
 
     output.clear();
 
     std::string xclbinPath;
     searchXsaAndDsa(m_idx, xsaXclbinPath, dsaXclbinPath, xclbinPath, output);
-    xclbinPath += xclbin;
-
-    if (stat(xrtTestCasePath.c_str(), &st) != 0 || stat(xclbinPath.c_str(), &st) != 0) {
-        //if bandwidth xclbin isn't present, skip the test
-        if(xclbin.compare("bandwidth.xclbin") == 0) {
-            output += "Bandwidth xclbin not available. Skipping validation.";
-            return -EOPNOTSUPP;
-        }
-        output += "ERROR: Failed to find ";
-        output += py;
-        output += " or ";
-        output += xclbin;
-        output += ", Shell package not installed properly.";
-        return -ENOENT;
-    }
+    std::string cmd;
+    
+    // if platform.json exists in the new shell pakcage then use the new testcase
+    // else fallback to old python test cases
+    
+    // NEW FLOW: runs if platform.json is available and the platform is not versal
+    // Note: we don't have cpp based versal bw testcase
+    auto json_exists = [xclbinPath]() 
+        {
+            return boost::filesystem::exists(xclbinPath + "platform.json") ? true : false;
+        };
+    if(json_exists() && xclbin.compare("versal_23_bandwidth.py") != 0) {
+        
+        //map old testcase names to new testcase names
+        static const std::map<std::string, std::string> test_map = {
+            { "22_verify.py",             "validate.exe"    },
+            { "23_bandwidth.py",          "kernel_bw.exe"   },
+            { "host_mem_23_bandwidth.py", "slavebridge.exe" }
+        };
+        
+        xrtTestCasePath += test_map.find(py)->second;
+        // in case the user is trying to run a new platform with an older XRT
```

Comment:
Could you clarify "older XRT“？

Reply:
By older XRT I mean any XRT before this PR, but I'll update the comment reflecting this

Reply:
> 
> 
> By older XRT I mean any XRT before this PR, but I'll update the comment reflecting this

you pr will make "older XRT" become "new XRT", so how do you handle "older XRT" case in your pr?

Reply:
So if we have a new shell (with platform.json) with an older XRT (without the .exe), then we'll throw an error saying: 
`ERROR: Failed to find verify.exe. Please update XRT.`